src/Content/Text/HTML.php

```diff
@@ -180,7 +180,9 @@ function ($matches) use (&$codeblocks) {
 		$xpath = new DomXPath($doc);
 		$list = $xpath->query("//pre");
 		foreach ($list as $node) {
-			$node->nodeValue = str_replace("\n", "\r", $node->nodeValue);
+			// Ensure to escape unescaped & - they will otherwise raise a warning
+			$safe_value = preg_replace('/&(?!\w+;)/', '&amp;', $node->nodeValue);
```

Comment:
What about any of the  `htmlencode`/`htmlentities` functions?

Reply:
These functions would be problematic, since we mustn't do this all the time, but we have to handle the case of bad HTML where this entity isn't encoded correctly.

Reply:
Ok.